pkg/monitors/filesystems/partitions_windows.go

```
@@ -0,0 +1,195 @@
+) ([]gopsutil.PartitionStat, error) {
+
+	stats := make([]gopsutil.PartitionStat, 0)
+	volNameBuf := make([]uint16, volumeNameBufferLength)
+
+	handle, err := findFirstVolume(&volNameBuf[0])
+	if err != nil {
+		return stats, errors.WithMessagef(err, "cannot find first-volume")
+	}
+	defer findVolumeClose(handle)
+
+	var volPaths []string
+	if volPaths, err = getVolumePaths(volNameBuf); err != nil {
+		return stats, errors.WithMessagef(err, "find paths error for first-volume %s", windows.UTF16ToString(volNameBuf))
+	}
+
+	var partitionStats []gopsutil.PartitionStat
+
+	if len(volPaths) > 0 {
+		partitionStats, err = getPartitionStats(getDriveType(volPaths[0]), volPaths, getFsNameAndFlags)
+		if err != nil {
+			return stats, errors.WithMessagef(err, "cannot find partition-stats for first-volume %s", windows.UTF16ToString(volNameBuf))
+		}
+		stats = append(stats, partitionStats...)
+	}
+
+	var lastError error
+	for {
+		volNameBuf = make([]uint16, volumeNameBufferLength)
+		if err = findNextVolume(handle, &volNameBuf[0]); err != nil {
+			if errno, ok := err.(syscall.Errno); ok && errno == windows.ERROR_NO_MORE_FILES {
+				break
+			}
+			lastError = errors.WithMessagef(err, "find next-volume last-error")
+			continue
+		}
+
+		volPaths, err = getVolumePaths(volNameBuf)
+		if err != nil {
+			lastError = errors.WithMessagef(err, "find paths last-error for volume %s", windows.UTF16ToString(volNameBuf))
+			continue
+		}
+
+		if len(volPaths) > 0 {
+			partitionStats, err = getPartitionStats(getDriveType(volPaths[0]), volPaths, getFsNameAndFlags)
+			if err != nil {
+				lastError = errors.WithMessagef(err, "find partition-stats last-error for volume %s", windows.UTF16ToString(volNameBuf))
+				continue
+			}
+			stats = append(stats, partitionStats...)
+		}
+	}
+
+	return stats, lastError
+}
+
+func getDriveType(rootPath string) (driveType uint32) {
+	rootPathPtr, _ := windows.UTF16PtrFromString(rootPath)
```

Are we swallowing a potential error here?

---

An error occurs only when the input string has null bytes according to https://pkg.go.dev/golang.org/x/sys@v0.0.0-20210823070655-63515b42dcdf/windows#UTF16PtrFromString. We are passing "good" input strings in all getDriveType() calls.